src/components/Response/TetrisGame.tsx

```diff
@@ -0,0 +1,391 @@
+        const blocks = this.store.blocks;
+
+        // Remove actual block from field to test for new insert position
+        tetrisGameGrid[yCoordinateOfActiveTile + blocks[activeBlockNumber][rotate][0][1]][xCoordinateOfActiveBlock + blocks[activeBlockNumber][rotate][0][0]] = 0;
+        tetrisGameGrid[yCoordinateOfActiveTile + blocks[activeBlockNumber][rotate][1][1]][xCoordinateOfActiveBlock + blocks[activeBlockNumber][rotate][1][0]] = 0;
+        tetrisGameGrid[yCoordinateOfActiveTile + blocks[activeBlockNumber][rotate][2][1]][xCoordinateOfActiveBlock + blocks[activeBlockNumber][rotate][2][0]] = 0;
+        tetrisGameGrid[yCoordinateOfActiveTile + blocks[activeBlockNumber][rotate][3][1]][xCoordinateOfActiveBlock + blocks[activeBlockNumber][rotate][3][0]] = 0;
+
+        // Test if the move can be executed on actual field
+        let xAddIsValid = true;
+        const shadowMap = getShadowBlock(tetrisGameGrid);
+        // check if block should move horizontally
+        if (xAdd !== 0) {
+            for (let block = 0; block < noOfBlock; block++) {
+                // Test if block can be moved without getting outside the board
+                if (
+                    xCoordinateOfActiveBlock + xAdd + blocks[activeBlockNumber][rotate][block][0] >= 0
+                    && xCoordinateOfActiveBlock + xAdd + blocks[activeBlockNumber][rotate][block][0] < Constants.BOARD_WIDTH
+                ) {
+                    if (tetrisGameGrid[yCoordinateOfActiveTile + blocks[activeBlockNumber][rotate][block][1]][xCoordinateOfActiveBlock
+                        + xAdd + blocks[activeBlockNumber][rotate][block][0]] !== 0) {
+                        // Prevent the move
+                        xAddIsValid = false;
+                    }
+                } else {
+                    // Prevent the move
+                    xAddIsValid = false;
+                }
+            }
+        }
+
+        // If horizontal move is valid update x variable (move the block)
+        if (xAddIsValid) {
+            xCoordinateOfActiveBlock += xAdd;
+        }
+
+        // Try to rotate the block
+        const maximumRotation = 3;
+        let newRotate = rotate + rotateAdd > maximumRotation ? 0 : rotate + rotateAdd;
+        let rotateIsValid = true;
+
+        // Test if block should rotate
+        if (rotateAdd !== 0) {
+            for (let block = 0; block < noOfBlock; block++) {
+                // Test if block can be rotated without getting outside the board
+                if (
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][0] >= 0 &&
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][0] <= Constants.BOARD_HEIGHT &&
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][1] >= 0 &&
+                    xCoordinateOfActiveBlock + blocks[activeBlockNumber][newRotate][block][1] <= Constants.BOARD_HEIGHT
```

Comment:
Looks incorrect. We should compare y-coordinate with board height. Also, blocks[activeBlockNumber][newRotate][block][1] represents y-coordinate for a tile.

Reply:
@abhinandan023 check this as well